demonhunter/nodes/manager/__init__.py

```diff
@@ -1,25 +1,43 @@
 from aiohttp import web
+import ssl
+import os
 
 from demonhunter.nodes.manager.app import add_routes, run_app
 from demonhunter.nodes.manager.agent.server import AgentManager
 from demonhunter.core.loggers.logfile import FileLogger
 
 
 class Manager:
-
     agents = list()
-    def __init__(self, loop, server_address, webapp_address, agent_manager=True,
-                 web_app=None, agent_password=None, logfile=None,
-                 port=16742):
+
+    def __init__(self, loop, server_address, webapp_address, port, certificate, key, agent_manager=True,
+                 web_app=None, agent_password=None, logfile=None):
         self.server_address = server_address
         self.webapp_address = webapp_address
         self.loop = loop
         self.port = port
+        self.certificate = certificate
+        self.key = key
+
+        # configure communication via ssl if the certificate and key are present
+        if self.certificate is not None and self.certificate is not None:
+            assert os.path.isfile(self.certificate)
+            assert os.path.isfile(self.key)
+            sc = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+            sc.load_cert_chain(self.certificate, self.key)
 
         if agent_manager:
+            # Set the values statically if nothing is specified
             if self.server_address is None:
                 self.server_address = "127.0.0.1"
-            coro = self.loop.create_server(lambda: AgentManager(self), self.server_address, self.port)
+            if self.port is None:
+                self.port = 16742
+            # check to see if key,cert are present and runs the corresponding server
+            if sc is not None:
```

Comment:
Yes, this would support the unsecure connection scheme going forward. I think that I've done it in such a way that would allow the module to function as it always has.